openverse_catalog/dags/providers/provider_api_scripts/smithsonian.py

```diff
@@ -186,6 +190,48 @@ def _gather_unit_sample(unit, sample_dir, retries=RETRIES, endpoint=SEARCH_ENDPO
             break
 
 
+def get_new_and_outdated_unit_codes(unit_code_set, sub_prov_dict=SUB_PROVIDERS):
+    sub_provider_unit_code_set = set()
+
+    for sub_prov, unit_code_sub_set in sub_prov_dict.items():
+        sub_provider_unit_code_set = sub_provider_unit_code_set.union(unit_code_sub_set)
```

Comment:
If I'm interpreting this correctly, I think we can just do something like the following:
```python
[ins] In [1]: d = {1: {1, 2, 3}, 2: {3, 4, 5}, 3: {8, 9}}

[ins] In [2]: set().union(*d.values())
Out[2]: {1, 2, 3, 4, 5, 8, 9}
```

```suggestion
    all_sub_provider_unit_codes = set().union(*sub_prov_dict.values())
```

Reply:
For sure -- this is just a copy/paste of the old logic 😄 I'll update the types while I'm at it!

Reply:
D'oh! Okay that makes a lot of sense :sweat_smile: